### ISSUE TITLE
TrustAllX509TrustManager lint warning with firebase-messaging 23.0.1

### ISSUE THREAD

**Issue by alvindizon** [firebase/firebase-android-sdk#3539]
### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: Android Studio Bumblebee | 2021.1.1 Patch 2
  * Firebase Component: Messaging (Database, Firestore, Storage, Functions, etc)
  * Component version: 23.0.1

### [REQUIRED] Step 3: Describe the problem
After upgrading firebase-messaging from 23.0.0 to 23.0.1, upon executing the Gradle `lintRelease` task, we encountered the following error on all of our modules (path modified to remove sensitive info):

```
/Users/me/dev/work/app/module/com/google/api/client/util/SslUtils$1.class: Error: checkClientTrusted is empty, which could cause insecure network traffic due to trusting arbitrary TLS/SSL certificates presented by peers [TrustAllX509TrustManager]

   Explanation for issues of type "TrustAllX509TrustManager":
   This check looks for X509TrustManager implementations whose
   checkServerTrusted or checkClientTrusted methods do nothing (thus trusting
   any certificate chain) which could result in insecure network traffic
   caused by trusting arbitrary TLS/SSL certificates presented by peers.

```
#### Steps to reproduce:
 Run `lintRelease` Gradle task

**Comment by argzdev** [comment]
Hi @alvindizon, thanks for reporting. I tried to replicate the issue using our [quickstart](https://github.com/firebase/quickstart-android) project, and then running `lintRelease` via Gradle task, however I didn
t experience any issues. Am I missing anything?

Could you provide more details, or an [MCVE](https://stackoverflow.com/help/minimal-reproducible-example)? It'll greatly help us in our investigation. Thanks!

**Comment by G00fY2** [comment]
We are also facing this issue:
```
13:12:27  > Task :app:lintDebug FAILED
13:12:27  /app/com/google/api/client/util/SslUtils$1.class: Error: checkClientTrusted is empty, which could cause insecure network traffic due to trusting arbitrary TLS/SSL certificates presented by peers [TrustAllX509TrustManager]
13:12:27  /app/com/google/api/client/util/SslUtils$1.class: Error: checkServerTrusted is empty, which could cause insecure network traffic due to trusting arbitrary TLS/SSL certificates presented by peers [TrustAllX509TrustManager]
13:12:27  
13:12:27     Explanation for issues of type "TrustAllX509TrustManager":
13:12:27     This check looks for X509TrustManager implementations whose
13:12:27     checkServerTrusted or checkClientTrusted methods do nothing (thus trusting
13:12:27     any certificate chain) which could result in insecure network traffic
13:12:27     caused by trusting arbitrary TLS/SSL certificates presented by peers.
13:12:27  
13:12:27  2 errors, 0 warnings
```

**Comment by wezley98** [comment]
Seeing the exact same error msg as others, BOM `29.1.0` -> `29.2.0`

Firebase SDKs in use by our app:

```
 const val FIREBASE_BOM = "com.google.firebase:firebase-bom:29.2.0"
 const val FIREBASE_CRASHLYTICS = "com.google.firebase:firebase-crashlytics-ktx"
 const val FIREBASE_ANALYTICS = "com.google.firebase:firebase-analytics-ktx"
 const val FIREBASE_PERF = "com.google.firebase:firebase-perf-ktx"
```

**Comment by alvindizon** [comment]
Hi @argzdev , thanks for answering, I can provide a few more details:

- we're not using BOM (but others have commented that they experienced it with BOM)
- we use AGP version 7.2.0-alpha06
-  minSdk is 24, targetSdk is 31 

Creating an MCVE is kind of difficult right now but I'll try if I have the time. For now we've silenced the warning by adding the following in our `lint.xml`:

```
   <issue id="TrustAllX509TrustManager" severity="ignore">
        <ignore path="*/com/google" />
    </issue>

```
The error points to `SslUtils`, which seems to be used by Google HTTP Client Library for Java (see repo and class [here](https://github.com/googleapis/google-http-java-client/blob/main/google-http-client/src/main/java/com/google/api/client/util/SslUtils.java)). I've filed an [issue](https://github.com/googleapis/google-http-java-client/issues/1603) in that repo as well

**Comment by gsakakihara** [comment]
This may have been fixed with the latest release. Could you try again with version 23.0.2 of the FCM SDK and see if you still see this issue?

**Comment by G00fY2** [comment]
I can confirm that this is fixed with BOM 29.2.1 (which includes Cloud Messaging 23.0.2). 👍

**Comment by argzdev** [comment]
Since this has been confirmed to be fixed with the latest version, I'll close this ticket now. Feel free to reopen if needed or add further questions. Thanks!

**Comment by alvindizon** [comment]
I can confirm that this has been fixed with 23.0.2 as well, thanks!
